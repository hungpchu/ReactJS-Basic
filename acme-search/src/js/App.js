import React from "react";
import ".././styles/App.css";
import Search from "./Search";

function App() {
  return (
    <div className="app-body">
      <h1 className="search-header">ACME Search</h1>
      <Search />
    </div>
  );
}

export default App;
