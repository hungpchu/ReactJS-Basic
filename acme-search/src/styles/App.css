input {
  font-size: 0.9rem;
  width: 250px;
  height: 20px;
  margin-right: 10px;
  margin-left: 140px;
  border-radius: 5px;
}

button {
  padding: 5px 5px;
  font-size: 0.9rem;
  text-align: center;
  cursor: pointer;
  background-color: whitesmoke;
  border-radius: 5px;
}

.app-body {
  overflow-x : auto; 
  overflow-y : auto; 
  background-color: #282c34;
  position: fixed;
  padding: 0;
  margin: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  /* justify-content: center; */
  font-size: 1.4rem;
  font-family: Arial, Helvetica, sans-serif;
  /* color: white; */
}

.search-header {
  color: #61dafb;
  text-align: center;
}
