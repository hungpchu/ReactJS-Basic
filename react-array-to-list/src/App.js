import React from 'react';
import logo from './logo.svg';
import './App.css';
import ArrayToList from './ArrayToList';
import ListAndKey from './ListAndKey';

function App() {
  return (
    <div className="App">
      <body className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <ArrayToList />
        <br/>
        <ListAndKey />
      </body>
    </div>
  );
}

export default App;
