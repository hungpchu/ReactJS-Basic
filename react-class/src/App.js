import React from 'react';
import logo from './logo.svg';
import './App.css';
import Member from './Member';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <Member name="Hung Chu" age="24" />
      </header>
    </div>
  );
}

export default App;
