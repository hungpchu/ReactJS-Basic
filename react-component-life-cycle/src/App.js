import React from 'react';
import logo from './logo.svg';
import './App.css';
// import CounterContinuous from './CounterContinuous.js';
import CounterAfter3 from './CounterAfter3Second';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        {/* <CounterContinuous /> */}
        <CounterAfter3 />
      </header>
    </div>
  );
}

export default App;
