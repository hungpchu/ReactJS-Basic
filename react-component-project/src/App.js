import React from 'react';
import './App.css';
import Header from './Header';
import Headline from './Headline';
import Aside from './Aside';
import ItemList from './ItemList';
import BoxList from './BoxList';
import Footer from './Footer';

function App() {
  return (
    <div id="layout">
      <Header />

      <main>
        <section id="content">
          <Headline />
           <ItemList />
        </section>

        <Aside />

        <BoxList />
      </main>

      <Footer />
    </div>
  );
}

export default App;
