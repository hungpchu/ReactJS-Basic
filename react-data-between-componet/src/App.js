import React from 'react';
import logo from './logo.svg';
import './App.css';

const member =
{
  text:"Hung Chu dep trai 45",
  info:{
    path: logo,
    name: 'Hung Chu',
    age: '24'
  }
}
// cách này truyền tham số mà không lồng function vào nhau
// function Avatar(props){
//   return (
//     <div className="avatar">
//       <img src={props.info.path} alt=""/>
//     </div>
//   );
// }

// function MemberInfo(props)
// {
//   return (
//     <div className="info">
//       {/* 
//        * props take infoObject base on here 
//        * <MemberInfo infoObject={member.info} />
//        */}
//       <p>{props.infoObject.name}</p>
//       <p>{props.infoObject.age}</p>
//     </div>
//   );
// }

// function Comment(props){
//   return(
//     <div className="comment">
//      {props.text} 
//     </div>
//   )
// };

// function Member()
// {
//   return (
//     <div className="member">
//       <Avatar info={member.info} />
//       <MemberInfo infoObject={member.info} />
//       <Comment text={member.text} />
//     </div>
//   );
// }

//  cách này truyền tham số mà  lồng function vào nhau
function Avatar(props){
  return(
    <div className="avatar">
      {/*
       * props sẽ lấy biến avatar object và lấy path 
       */}
      <img src={props.avatar.path} alt="" />
    </div>
  )
};

function MemberInfo(props){
  return(
    <div className="info">
      {/* 
       * avatar là tên biến truyền vào Avatar function
       */}
      <Avatar avatar={props.info} />
      <h2>{props.info.name}</h2>
      <p>Tuổi: {props.info.age}</p>
    </div>
  )
};

function Comment(props){
  return(
    <div className="comment">
      {props.text}
    </div>
  )
};

function Member(){
  return(
    <div className="member">
      {/* 
        * info là biến truyền vào MemberInfo function
        */}
      <MemberInfo info={member.info} />
      {/* 
        * text là biến truyền vào Comment function
        */}
      <Comment text={member.text} />
    </div>
  )
};


export default Member;
