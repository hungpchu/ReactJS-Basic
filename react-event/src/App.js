import React from 'react';
import logo from './logo.svg';
import './App.css';
import Button from './Button.js';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <div id="hung"></div> 
        <Button />
      </header>
    </div>
   
  );
}

export default App;
