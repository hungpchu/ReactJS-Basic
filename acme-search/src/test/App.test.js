import React from 'react';
import renderer from 'react-test-renderer';
import App from './../js/App.js';

describe('<App />', () => {

  describe('(snapshot)', () => {
    it('renders the component correctly', () => {
    const appSnapShot = renderer
      .create(<App />)
      .toJSON();
    expect(appSnapShot).toMatchSnapshot();
  });
  })
})