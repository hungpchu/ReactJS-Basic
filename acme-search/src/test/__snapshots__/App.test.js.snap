// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> (snapshot) renders the component correctly 1`] = `
<div
  className="app-body"
>
  <h1
    className="search-header"
  >
    ACME Search
  </h1>
  <div
    className="seach-body"
  >
    <div
      className="seach-box"
    >
      <input
        className="search-input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search"
        type="text"
      />
      <button
        className="search-button"
        onClick={[Function]}
      >
        Search
      </button>
      <button
        id="update-data-button"
        onClick={[Function]}
      >
        Update Data
      </button>
    </div>
  </div>
</div>
`;
